Giraf.IntegrationTests/Endpoints/CitizensEndpointTests.cs

```diff
@@ -169,30 +207,35 @@ public async Task AddCitizen_ReturnsOk_WhenOrganizationExists()
             var client = factory.CreateClient();
 
             // Get the organization ID
-            using (var scope = factory.Services.CreateScope())
-            {
-                var dbContext = scope.ServiceProvider.GetRequiredService<GirafDbContext>();
-                var organization = await dbContext.Organizations.FirstOrDefaultAsync();
-                Assert.NotNull(organization);
-                var organizationId = organization.Id;
+            var scope = factory.Services.CreateScope();
+            var dbContext = scope.ServiceProvider.GetRequiredService<GirafDbContext>();
+            var organization = await dbContext.Organizations.FirstOrDefaultAsync();
+            Assert.NotNull(organization);
+            var organizationId = organization.Id;
 
-                var createCitizenDto = new CreateCitizenDTO("New", "Citizen");
+            var createCitizenDto = new CreateCitizenDTO("New", "Citizen");
 
-                // Act
-                var response = await client.PostAsJsonAsync($"/citizens/{organizationId}/add-citizen", createCitizenDto);
+            var testOrgId = 1;
```

Comment:
There's no reason to use a mock testId here. On line 214, you extract the actual organizationId - use that for the claim.